govcd/edgegateway.go

```diff
@@ -145,15 +145,22 @@ func (eGW *EdgeGateway) AddDhcpPool(network *types.OrgVDCNetwork, dhcppool []int
 
 }
 
-// Temporary fix for existing Remove functions as it doesn't support network interfaces
-// TODO: remove this function when ruleId is available
-func (eGW *EdgeGateway) RemoveNATMappingRule(networkHref, natType, externalIP, internalIP, port string) (Task, error) {
-	return eGW.RemoveNATPortRule(networkHref, natType, externalIP, port, internalIP, port)
+// Deprecated in favor of RemoveNATRuleAsync, RemoveNATRule
```

Comment:
I risk to be annoying but I just read the suggested deprecation string and I think it's great opportunity to use it as there may be some tooling that is aware of it. Added an example below from standard lib.
It still contains function documentation and then a `Deprecated: followed by some information about the deprecation`


https://blog.golang.org/godoc-documenting-go-code
> Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with "Deprecated:" followed by some information about the deprecation. There are a few examples in the standard library. 

```go
// Copy returns a new Regexp object copied from re.
// Calling Longest on one copy does not affect another.
//
// Deprecated: In earlier releases, when using a Regexp in multiple goroutines,
// giving each goroutine its own copy helped to avoid lock contention.
// As of Go 1.12, using Copy is no longer necessary to avoid lock contention.
// Copy may still be appropriate if the reason for its use is to make
// two copies with different Longest settings.
func (re *Regexp) Copy() *Regexp {
	re2 := *re
	return &re2
}
```

Here's how it looks in the generated documentation: https://golang.org/pkg/regexp/#Regexp.Copy

Reply:
Fixed